### PR TITLE
refactor: Replace _.parseInt with Number

### PR DESCRIPTION
**Pull request by jishnu-mohan**
Replace lodash _.parseInt with native Number.

Addresses: #7747

### CODE REVIEW

File: lib/plugins/aws/package/compile/functions/index.js [hunk 1]
```diff
@@ -398,7 +398,7 @@ class AwsCompileFunctions {
 
       if (functionObject.reservedConcurrency || functionObject.reservedConcurrency === 0) {
         // Try convert reservedConcurrency to integer
-        const reservedConcurrency = _.parseInt(functionObject.reservedConcurrency);
+        const reservedConcurrency = Number(functionObject.reservedConcurrency);
```
Comment (safv12) [inline lib/plugins/aws/package/compile/functions/index.js:401]: Just for know, Why not to use `Number.parseInt(x, base)`

https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/parseInt

Comment (jishnu-mohan) [reply lib/plugins/aws/package/compile/functions/index.js:401]: The only difference I could find about `Number()` and `Number.parseInt` is that `Number.parseInt(null)` returns `NaN` and `Number(null)` returns 0. So, I think it's safe to use `Number.parseInt`. Thanks @safv12 for pointing it out.

Comment (medikoo) [reply lib/plugins/aws/package/compile/functions/index.js:401]: `parseInt` is for parsing numbers out of strings. Technically logic is as:

```
resolveIntegerPrefixOutOfString(coerceToString(value))
```

While what `Number` does is:

```
coerceToNumber(value)
```

Here we expect user to input number type directly (and not string with integer prefix), so using `parseInt` seems as not perfectly justified and it'll also hide configuration error if someone puts `1.5`

As a side note input validation will shortly be handled by JSON schema based validation (see https://github.com/serverless/serverless/issues/6562) if we would have that already in, then best would be to just assume that `functionObject.reservedConcurrency` is good as is (as it's already validated), but until is the case, let's use `Number`

**Comment by medikoo** [comment]
Closing, as finally it was addressed with #7877 

Anyway @jishnu-mohan great thanks, for attempting to address that!

**Pull request closed without merging**